app/src/components/common/EmailInputWithDomainBasedSuggestions/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import _ from "lodash";
+import { getFunctions, httpsCallable } from "firebase/functions";
+import { useSelector } from "react-redux";
+import { getUserAuthDetails } from "store/selectors";
+import { getDomainFromEmail, isCompanyEmail, isEmailValid } from "utils/FormattingHelper";
+import CreatableSelect from "react-select/creatable";
+import { MultiValue } from "react-select";
+
+interface Props {
+  onChange: (emails: string[]) => void;
+  transparentBackground?: boolean;
+}
+
+const EmailInputWithDomainBasedSuggestions: React.FC<Props> = ({ onChange, transparentBackground = false }) => {
+  const user = useSelector(getUserAuthDetails);
+  const userEmail = user?.details?.profile?.email;
+
+  const [allSuggestions, setAllSuggestions] = useState<string[]>([]);
+
+  const getOrganizationUsers = useMemo(() => httpsCallable(getFunctions(), "users-getOrganizationUsers"), []);
+
+  useEffect(() => {
+    if (!isCompanyEmail(userEmail)) return;
+
+    getOrganizationUsers({ domain: getDomainFromEmail(userEmail) }).then((res: any) => {
+      const users = res.data.users;
+      const emails = users.map((user: any) => user.email);
+      setAllSuggestions(_.without(emails, userEmail));
+    });
+  }, [getOrganizationUsers, userEmail]);
+
+  const optionsFromSuggestions = useMemo(() => {
+    return allSuggestions.map((suggestion) => {
+      return {
+        label: suggestion,
+        value: suggestion,
+      };
+    });
+  }, [allSuggestions]);
```

Comment:
Can we avoid this memoization? Instead set this mapped data to `allSuggestions` state variable after getting organization users